---
title: "Running a retrotranscriptome-wide association study (rTWAS)"
author: "by Rodrigo R. R. Duarte (rodrigo.duarte@kcl.ac.uk)"
date: "2024-04-01"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## <b>1) Set up a conda environment to run FUSION and FOCUS. </b>
Installation time: approximately 5-10 min, depending on your machine configuration. <br>
Set up the FUSION program folder: <br>
`cd ~/scratch/programs/`<br>
`wget https://github.com/gusevlab/fusion_twas/archive/master.zip -O fusion.zip`<br>
`unzip fusion.zip`

The following assumes that you already have conda installed. For more information on conda, see https://docs.conda.io/en/latest/miniconda.html <br>
To create the conda environment for fusion and focus, please using the yml files provided - see https://github.com/rodrigoduarte88/neuro_rTWAS/blob/main/fusion_final_environment.yml <br>
`conda env create --file fusion_final_environment.yml` <br>
This yml file contains most software and library versions required to run focus/fusion.<br>

We still will need to install the R library "plink2R". To do this, rename libraries for plink2R in the conda environment folder (as detailed [here](https://stackoverflow.com/questions/64541476/cant-load-librlapack-so-librblas-so-under-conda)).<br>
`cd /users/rodrigoduarte88/scratch/miniconda3/envs/fusion_final/lib`<br>
`mv liblapack.so libRlapack.so`<br>
`mv libblas.so libRblas.so`<br>
<br>
You will need to start R, and install manually plink2R using the following command: <br>
`conda activate fusion_final` <br>
`R` <br>
`devtools::install_github("carbocation/plink2R/plink2R", ref="carbocation-permit-r361")` <br><br>


Now, let's create the conda environment for FOCUS<br>
`conda create -n focus python=3.7 r-base` <br>
`conda activate focus` <br>
`pip install pyfocus --user` <br>
`pip install mygene --user` <br>
`pip install rpy2 --user` <br>


## <b>2) Download the required files and decompress</b>
These include the SNP weights for FOCUS/FUSION and the 1000 Genomes reference panel for the population of interest. Please download the required files from the King's College London Research Data Repository (KORDS), at https://doi.org/10.18742/22179655. Then decompress files.<br>
`tar zxvf FOCUS_weights.tgz`<br>
`tar zxvf FUSION_weights.tgz`<br>
`tar zxvf 1000G_ref_panel.tgz`<br>
<br>
N.B.: The reference panels are annotated with dbsnp151/hg19 information.

## <b>3) Preprocessing GWAS summary statistics</b>
Your GWAS summary statistics must be annotated with variant IDs according to dbsnp151. Use munge_sumstats.py from the [ldsc](https://github.com/bulik/ldsc) package for pre-filtering. You can find an example of how this was done on the scripts available from https://github.com/rodrigoduarte88/TWAS_HERVs-SCZ. You can also check the [FUSION guidelines](http://gusevlab.org/projects/fusion/) for additional instructions.


Summary statistics for FUSION should look like:

```
SNP     A1      A2      Z
rs10    A       C       -0.501
rs1000000       G       A       2.238
rs10000003      A       G       -1.324
rs10000010      T       C       -0.082
rs10000013      C       A       -2.04
```

Summary statistics for FOCUS should look like:

```
CHR     SNP     BP      A1      A2      Z       N
7       rs10    92383888        A       C       -0.501  58749.13
12      rs1000000       126890980       G       A       2.238   58749.13
4       rs10000003      57561647        A       G       -1.324  58749.13
4       rs10000010      21618674        T       C       -0.082  58749.13
4       rs10000013      37225069        C       A       -2.04   58749.13
```

## <b>4) Running [FUSION](http://gusevlab.org/projects/fusion/)</b>
To run FUSION, activate the conda environment, and use the FUSION weights and linkage disequilibrium reference panel provided. <br>
`conda activate fusion_final`<br>
<br>
`Rscript FUSION.assoc_test.R \`<br>
`--sumstats PGC2.SCZ.sumstats.fusion \`<br>
`--weights ./wrapped/CMC.pos \`<br>
`--weights_dir ./wrapped/ \`<br>
`--ref_ld_chr ./LDREF_harmonized/1000G.EUR. \`<br>
`--chr 22 \`<br>
`--out PGC2.SCZ.22.dat` <br>
<br>
To run the conditional analysis, you can follow the instructions as provided by the authors of FUSION. For example, first, you have to obtain a file containing only Bonferroni significant hits, and then you can perform the conditional analysis. <br>
<br>
Combine all files from all chromosomes <br>
`head -1  PGC2.SCZ.1.dat > SCZ_____all_chr.tsv`<br>
`tail -n +2 -q PGC2.SCZ.* >> SCZ_____all_chr.tsv`<br>
<br>
Create file with significant hits only (Bonferroni) <br>
`bonferroni_p='bc -l <<< "scale=50; 0.05/8212"'` # 8212 is the number of expressed features in the weights<br>
`cat SCZ_____all_chr.tsv | awk -v var="${bonferroni_p}" 'NR == 1 || $20 < var' > SCZ_____all_chr.tsv.Sig`<br>
<br>
`Rscript FUSION.post_process.R \`<br>
`--sumstats PGC2.SCZ.sumstats.fusion \`<br>
`--input SCZ_____all_chr.tsv.Sig \`<br>
`--out SCZ_____all_chr.tsv.Sig.analysis \`<br>
`--ref_ld_chr ./LDREF_harmonized/1000G.EUR. \`<br>
`--chr 22 \`<br>
`--plot --locus_win 100000`<br>

## <b>5) Running [FOCUS](https://github.com/bogdanlab/focus)</b>
To run FOCUS, activate the conda environment, and use the FOCUS weights and linkage disequilibrium reference panel provided. <br>
`conda activate focus` <br>
`module load mesa-glu/9.0.1-gcc-9.4.0` # this is for CREATE users - loads libGL.so.1<br>
<br>
`focus finemap schizophrenia.gwas.focus \`<br>
`LDREF_harmonized/1000G.EUR.22 CMC_brain_focus_database.db \`<br>
`--chr 22 --plot --p-threshold 5E-08 \`<br>
`--out SCZ_pgc3_CMC.5e-8.chr.22 --locations 37:EUR`<br>
<br>
<br>
<br>

For interpretation of the output files, please use the instructions provided by the authors of [FOCUS](https://github.com/bogdanlab/focus) and [FUSION](http://gusevlab.org/projects/fusion/). The results contain gene and HERV expression signatures associated with genetic susceptibility to your trait of interest.
<br>
<br>
<br>





