---
title: "Running a retrotranscriptome-wide association study (rTWAS)"
author: "by Rodrigo R. R. Duarte"
date: "2023-02-26"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## <b>1) Set up the environment</b>
#### Set up the FUSION program folder
`cd ~/scratch/programs/`<br>
`wget https://github.com/gusevlab/fusion_twas/archive/master.zip -O fusion.zip`<br>
`unzip fusion.zip`

#### Create conda environment using the yml file provided
`conda env create --file fusion_final_environment.yml`

#### For KCL's HPC CREATE, you have to create shortcuts for plink2r libraries (otherwise installation in R next won't succeed) # see https://github.com/gusevlab/fusion_twas/issues/13#issuecomment-1202150944
`cd ~/scratch/miniconda3/envs/fusion_final/lib`<br>
`mv liblapack.so libRlapack.so`<br>
`mv libblas.so libRblas.so`


## <b>2) Download SNP weights for FOCUS/FUSION and the 1000 Genomes reference panel</b>
`wget ________________`<br>
`wget ________________`<br>
`wget ________________`<br>
N.B.: Reference panel corresponds to the European subset of the 1,000 Genomes project annotated with dbsnp151/hg19.

## <b>3) Preprocessing GWAS summary statistics</b>
Your GWAS summary statistics must be annotated with variant IDs according to dbsnp151. Use munge_sumstats.py from the [ldsc](https://github.com/bulik/ldsc) package for pre-filtering. You can find an example of how this was done on the scripts available from https://github.com/rodrigoduarte88/TWAS_HERVs-SCZ. You can also check the [FUSION guidelines](http://gusevlab.org/projects/fusion/) for additional instructions.


Summary statistics for FUSION should look like:

```
SNP     A1      A2      Z
rs10    A       C       -0.501
rs1000000       G       A       2.238
rs10000003      A       G       -1.324
rs10000010      T       C       -0.082
rs10000013      C       A       -2.04
```

Summary statistics for FOCUS should look like:

```
CHR     SNP     BP      A1      A2      Z       N
7       rs10    92383888        A       C       -0.501  58749.13
12      rs1000000       126890980       G       A       2.238   58749.13
4       rs10000003      57561647        A       G       -1.324  58749.13
4       rs10000010      21618674        T       C       -0.082  58749.13
4       rs10000013      37225069        C       A       -2.04   58749.13
```

## <b>4) Before running FUSION, switch on the conda environment, and proceed with the [FUSION analysis](http://gusevlab.org/projects/fusion/) as recommended by the authors</b>
`conda activate fusion_final`
<br>
E.g.,:
```
Rscript FUSION.assoc_test.R \
--sumstats PGC2.SCZ.sumstats \
--weights ./WEIGHTS/GTEx.Whole_Blood.pos \
--weights_dir ./WEIGHTS/ \
--ref_ld_chr ./LDREF/1000G.EUR. \
--chr 22 \
--out PGC2.SCZ.22.dat
```








